### PR TITLE
Rust single cargo

### PR DESCRIPTION
**Pull request by Cypher1**
Changes:
- merges gtk-rust-* crates
- introduces features: one for the 'host' and one for the 'modules' to enable optional dependencies
- removes run.sh usage of --manifest-path as it is only supported by some tools and this can be confusing
- introduces a 'common' module to enable shared code, gated by features
- removes duplicated struct and cptr definitions

### CODE REVIEW

**Comment by Cypher1** [comment]
@mykmartin to review?

File: run.sh [hunk 1]
```diff
@@ -15,17 +15,14 @@
 # limitations under the License.
 #
 
-set -e
+set -ex
```
Comment (mykmartin) [inline run.sh]: I'd prefer not to have `-x` on by default.

**Review by mykmartin — APPROVED** [review]
Nice!

File: run.sh (hunk 1 above)
Comment (Cypher1) [reply run.sh]: Done! (Via phone :O)
I love it for debugging but it is pretty noisey

**Comment by Cypher1** [comment]
@tiziano88 any chance I could get merge privileges?

**Comment by tiziano88** [comment]
> 

Done (I think). anyways I believe @mykmartin should also be able to add people as writers / maintainers.

**Comment by Cypher1** [comment]
Thanks!

**Pull request merged**